packages/v2/gui/src/kodeverk/context/KodeverkContext.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { createContext, useMemo, useState } from 'react';
+import { type AlleKodeverk } from '@k9-sak-web/lib/types/index.js';
+
+type KodeverkContextValuesType = {
+  behandlingType: string | undefined;
+  kodeverk: AlleKodeverk | undefined;
+  klageKodeverk: AlleKodeverk | undefined;
+  tilbakeKodeverk: AlleKodeverk | undefined;
+};
+
+type KodeverkContextFunctionsType = {
+  setKodeverkContext?: ({
+    behandlingType,
+    kodeverk,
+    klageKodeverk,
+    tilbakeKodeverk,
+  }: {
+    behandlingType: string;
+    kodeverk: AlleKodeverk;
+    klageKodeverk: AlleKodeverk;
+    tilbakeKodeverk: AlleKodeverk;
+  }) => void;
+};
+
+type KodeverkContextType = KodeverkContextValuesType & KodeverkContextFunctionsType;
+
+const initialValue: KodeverkContextValuesType = {
+  behandlingType: undefined,
+  kodeverk: undefined,
+  klageKodeverk: undefined,
+  tilbakeKodeverk: undefined,
+};
+
+const isEmptyObject = (obj: any) => obj && Object.keys(obj).length === 0 && obj.constructor === Object;
+
+export const KodeverkContext = createContext<KodeverkContextType>(initialValue);
+
+export const KodeverkProvider = ({
+  children,
+  behandlingType = undefined,
+  kodeverk = undefined,
+  klageKodeverk = undefined,
+  tilbakeKodeverk = undefined,
+}: {
+  children: React.ReactNode;
+  behandlingType: string | undefined;
+  kodeverk?: AlleKodeverk;
+  klageKodeverk?: AlleKodeverk;
+  tilbakeKodeverk?: AlleKodeverk;
+}) => {
+  const [kodeverkContext, setKodeverkContextState] = useState<KodeverkContextValuesType>(initialValue);
+
+  /**
+   *
+   * Verifiser at verdiene er satt og ikke er like før oppdatering av state
+   */
+  const setKodeverkContext = ({
+    behandlingType: newBehandlingType,
+    kodeverk: newKodeverk,
+    klageKodeverk: newKlageKodeverk,
+    tilbakeKodeverk: newTilbakeKodeverk,
+  }: KodeverkContextValuesType) => {
+    if (
+      behandlingType !== newBehandlingType ||
+      (!isEmptyObject(newKodeverk) && kodeverk !== newKodeverk) ||
+      (!isEmptyObject(newKlageKodeverk) && klageKodeverk !== newKlageKodeverk) ||
+      (!isEmptyObject(newTilbakeKodeverk) && tilbakeKodeverk !== newTilbakeKodeverk)
+    ) {
+      setKodeverkContextState({
+        behandlingType: newBehandlingType,
+        kodeverk: newKodeverk,
+        klageKodeverk: newKlageKodeverk,
+        tilbakeKodeverk: newTilbakeKodeverk,
+      });
+    }
+  };
+
+  const value = useMemo(
```

Comment:
Er det noko verdi i å bruke useMemo her? Ser ikkje ut til at "factory" funksjonen gjere noko tung jobb.

Reply:
Det var i forbindelse med å unngå re-rendring ja. Men jeg er ikke sikker på at den trengs lenger for å være ærlig.

Reply:
@josstn  Jo, den burde være memoet, eller tenker jeg feil?

Reply:
Viss det er nødvendig for å unngå noko uønska re-rendering, så burde den vel vere det (eg har ikkje testa noko).

Men eg trur ikkje det er noko vits å ha den der mtp ytelse, som vel er det useMemo i første omgang er tenkt å fikse.